### PR TITLE
Make test "tuned profile validation" more flexible

### PR DESCRIPTION
**Pull request by dshchedr**
This PR improves the test in scenario when we have some additional Tuned based on PAO profile:

```
    [test_id:28467][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Should contain configuration injected through the openshift-node-performance profile [It]
    /home/dshchedr/job/cnf/sno/performance-addon-operators/functests/1_performance/performance.go:226
    Timed out after 480.000s.
    active_profile is not set to openshift-node-performance-performance. <nil>
    Expected
        <string>: openshift-configuration-hotfixes
    to equal
        <string>: openshift-node-performance-performance

``` 
This happens because the file /etc/tuned/active_profile on the node contains new Tuned name (openshift-configuration-hotfixes), but we are trying to compare it with name in Performance Profile

Small improvement in this PR does verification if some another Tuned is based on PAO profile and changes name accordingly.

### CODE REVIEW

**Comment by fromanirh** [comment]
/lgtm
/hold
looks good to me, but I'd like Artyom's review as well

File: functests/1_performance/performance.go [hunk 1]
```diff
@@ -781,6 +782,19 @@ func validatTunedActiveProfile(nodes []corev1.Node) {
 	var err error
 	var out []byte
 	activeProfileName := components.GetComponentName(testutils.PerformanceProfileName, components.ProfileNamePerformance)
+
+	// check if some another Tuned profile overwrites PAO profile
+	tunedList := &tunedv1.TunedList{}
+	if err := testclient.Client.List(context.TODO(), tunedList); err != nil {
+		println(err)
```
Comment (cynepco3hahue) [inline functests/1_performance/performance.go:789]: What does `println` is doing and where does it define?

File: functests/1_performance/performance.go [hunk 2]
```diff
@@ -781,6 +782,19 @@ func validatTunedActiveProfile(nodes []corev1.Node) {
 	var err error
 	var out []byte
 	activeProfileName := components.GetComponentName(testutils.PerformanceProfileName, components.ProfileNamePerformance)
+
+	// check if some another Tuned profile overwrites PAO profile
```
Comment (cynepco3hahue) [inline functests/1_performance/performance.go:786]: Why not skip the test in this case?

File: functests/1_performance/performance.go (hunk 1 above)
Comment (fromanirh) [reply functests/1_performance/performance.go:789]: and why the test line passed if this is undefined?

Comment (fromanirh) [reply functests/1_performance/performance.go:789]: TIL: https://golang.org/pkg/builtin/#println https://play.golang.org/p/8ltQKd03z9X

Comment (cynepco3hahue) [reply functests/1_performance/performance.go:789]: hm `Println is useful for bootstrapping and debugging; it is not guaranteed to stay in the language.`, let's just use klog it anyway already used under the method

Comment (fromanirh) [reply functests/1_performance/performance.go:789]: agreed :) at least now we knw why it compiled fine :)

Comment (dshchedr) [reply functests/1_performance/performance.go]: to be honest, now I can't find from a which corner of my head I've taken this... 

changed to `klog.Error(err)`

File: functests/1_performance/performance.go (hunk 2 above)
Comment (dshchedr) [reply functests/1_performance/performance.go:786]: this function _validatTunedActiveProfile()_ is a part of several tests where we are also verifying some system configurations. Most of users should not change these configurations, so I'm not sure skipping the whole test is a good idea.. User will be warned that this test may fail, but in most of scenarios it should be passed.

File: functests/1_performance/performance.go [hunk 3]
```diff
@@ -781,6 +782,19 @@ func validatTunedActiveProfile(nodes []corev1.Node) {
...
+	tunedList := &tunedv1.TunedList{}
+	if err := testclient.Client.List(context.TODO(), tunedList); err != nil {
+		klog.Error(err)
+	}
+	for _, t := range tunedList.Items {
+		if strings.Contains(*t.Spec.Profile[0].Data, fmt.Sprintf("include=%s", activeProfileName)) {
```
Comment (cynepco3hahue) [inline functests/1_performance/performance.go:792]: we should check:
- `len(t.Spec.Profile) > 0`
- `t.Spec.Profile[0].Data != nil`

File: functests/1_performance/performance.go [hunk 4]
```diff
@@ -781,6 +782,19 @@ func validatTunedActiveProfile(nodes []corev1.Node) {
...
+			klog.Warning(fmt.Sprintf("PAO tuned profile amended by '%s' profile, test may fail", t.Name))
+			activeProfileName = *t.Spec.Profile[0].Name
```
Comment (cynepco3hahue) [inline functests/1_performance/performance.go:794]: the same check that `t.Spec.Profile[0].Name != nil` before dereferece

File: functests/1_performance/performance.go [hunk 5]
```diff
@@ -781,6 +782,19 @@ func validatTunedActiveProfile(nodes []corev1.Node) {
 	var err error
 	var out []byte
 	activeProfileName := components.GetComponentName(testutils.PerformanceProfileName, components.ProfileNamePerformance)
+
+	// check if some another Tuned profile overwrites PAO profile
+	tunedList := &tunedv1.TunedList{}
+	if err := testclient.Client.List(context.TODO(), tunedList); err != nil {
+		klog.Error(err)
+	}
+	for _, t := range tunedList.Items {
```
Comment (cynepco3hahue) [inline functests/1_performance/performance.go:791]: if you get an error, probably tunedList will be nil

Comment (dshchedr) [reply functests/1_performance/performance.go:791]: error should not happen, added: 	_Expect(err).NotTo(HaveOccurred())_

File: functests/1_performance/performance.go (hunk 3 above)
Comment (dshchedr) [reply functests/1_performance/performance.go]: added this checking

File: functests/1_performance/performance.go (hunk 4 above)
Comment (dshchedr) [reply functests/1_performance/performance.go]: hm, not sure if this object can be without _Name_ field, but added checking that.. thanks

**Pull request merged**